posthog/warehouse/api/saved_query.py

```diff
@@ -158,6 +156,10 @@ def destroy(self, request: request.Request, *args: Any, **kwargs: Any) -> respon
         instance: DataWarehouseSavedQuery = self.get_object()
         DataWarehouseJoin.objects.filter(source_table_name=instance.name).delete()
         DataWarehouseJoin.objects.filter(joining_table_name=instance.name).delete()
+
+        if instance.table is not None:
+            instance.table.soft_delete()
+
         self.perform_destroy(instance)
```

Comment:
This `self.perform_destroy(instance)` will hard delete the saved query and underlying table FYI. So, probably no need to do the soft delete on the table - we currently don't soft delete saved queries which is probably fine for now?!

Reply:
How does it delete the underlying table? SavedQuery now has a FK to table but there's no reverse relationship.